controllers/rpaasinstance_controller.go

```diff
@@ -141,31 +174,59 @@ func (r *RpaasInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Nginx CRD
 	nginx := newNginx(instanceMergedWithFlavors, plan, configMap)
-	if err = r.reconcileNginx(ctx, instanceMergedWithFlavors, nginx); err != nil {
+	changes["nginx"], err = r.reconcileNginx(ctx, instanceMergedWithFlavors, nginx)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err = r.reconcileTLSSessionResumption(ctx, instanceMergedWithFlavors); err != nil {
+	// Session Resumption
+	changes["sessionResumption"], err = r.reconcileTLSSessionResumption(ctx, instanceMergedWithFlavors)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err = r.reconcileHPA(ctx, instanceMergedWithFlavors, nginx); err != nil {
+	// HPA
+	changes["hpa"], err = r.reconcileHPA(ctx, instanceMergedWithFlavors, nginx)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err = r.reconcilePDB(ctx, instanceMergedWithFlavors, nginx); err != nil {
+	// PDB
+	changes["pdb"], err = r.reconcilePDB(ctx, instanceMergedWithFlavors, nginx)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err = r.refreshStatus(ctx, instance, nginx); err != nil {
+	if listOfChanges := listChanges(changes); len(listOfChanges) > 0 {
+		if systemRollout {
+			r.EventRecorder.Eventf(instance, corev1.EventTypeWarning, "RpaasInstanceSystemRolloutApplied", "RPaaS controller has updated these resources: %s to ensure system consistency", strings.Join(listOfChanges, ", "))
+		}
+	} else {
+		reservation.Cancel()
+	}
+
+	if err = r.refreshStatus(ctx, instance, instanceHash, nginx); err != nil {
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *RpaasInstanceReconciler) refreshStatus(ctx context.Context, instance *v1alpha1.RpaasInstance, newNginx *nginxv1alpha1.Nginx) error {
+func listChanges(changes map[string]bool) []string {
```

Comment:
being a bit picky here, but could it be `getChangesList`?
`listChanges` makes it sound like this function is responsible for the listing